plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/LoginServiceUi.java

```diff
@@ -126,16 +131,23 @@ private String showProgressDialogAndWaitForCode(final String message,
       final IOException[] exceptionHolder = new IOException[1];
       final Semaphore wait = new Semaphore(0 /* initially zero permit */);
 
-      new ProgressMonitorDialog(shellProvider.getShell()) {
+      final Shell parentShell = shellProvider.getShell();
+      new ProgressMonitorDialog(parentShell) {
         @Override
         protected void configureShell(Shell shell) {
           super.configureShell(shell);
           shell.setText(Messages.LOGIN_PROGRESS_DIALOG_TITLE);
+
+          AnalyticsPingManager.getInstance().sendPing(
+              AnalyticsEvents.LOGIN_START, null, null, shell);
```

Comment:
A minor quibble, but a constructor doesn't seem the right place for this event.  Why not either within the run() or just above?